### DISCUSSION TITLE
Nested <tr> component disappears on action call

### DISCUSSION THREAD

**Discussion by krizzdev** [livewire/livewire#3097] [7. Bugs (archived)]
As suggested in #36 I open a new issues with a similar problem.

### Description
I have multiple tr child components in a loop (flagged with a :key of course). When I call an action inside of the Child Component view the whole component disappears. (Note: the child blade view isn’t wrapped with a div, but with a tr)

### Exact steps to reproduce
This whole bug appears when using livewire and vue together (with the livewire-vue plugin). When I remove the livewire-vue script it works again. But I need the livewire-vue extension for other views inside of my project, so I can’t remove it
Use: vue / livewire / livewire-vue and add nested table-item components with actions. 

Create a parent livewire table component.

Create a child tr livewire component (not wrapped with div)
Add some wire:click or wire:model actions on the child blade view.

Here is a playground example with the bug: 
https://laravelplayground.com/#/snippets/b6d0a43f-e6f5-441d-a2a8-dfe0bc535624

Here a version without the livewire-vue plugin (where it’s working) 
https://laravelplayground.com/#/snippets/6e7ff756-ec1b-4880-8a5b-cc81f3386577

Note: I can't wrap the "tr" with "div" because otherwise the child component tries to call the parent methods.

Unfortunately I don't have the js devtool skills yet so maybe someone can help me out :D 

Kind Regards
Christian

**Context**
Livewire 2.3.5 
livewire-vue 0.3.1
AlpineJs 2.7.3

**Comment by Thommis** [comment]
same here!

**Comment by krizzdev** [comment]
It's because of some dom manipulations inside of the livewire/vue index.js  hook.

It removes all td's and the root tr element.

```
 const div = document.createElement('div')
    div.innerHTML =  message.response.effects.html

    new window.Vue().$mount(div.firstElementChild)

    message.response.effects.html = div.firstElementChild.outerHTML
```
    
 A workaround for me is to add a:
 
```
 if (message.response.effects.html.startsWith("<tr")) {
        return;
    }
```
    
  in the hook.
   But with this it won't be mounted to the Vue component. For me it has no issues. But for others it may break their apps.
   Maybe someone can improve this?

**Comment by nathanjansen** [comment]
I am currently not able to write an extended answer.
Did you checkout wire:key (https://laravel-livewire.com/docs/2.x/troubleshooting)?

**Comment by krizzdev** [comment]
Thanks for your response. Yes tried it on different html tags and my livewire components are flagged also with unique wire:keys. Didn't help.

**Comment by thyseus** [comment]
I can confirm that there seems still to be a problem in livewire ( using 2.3.6) with this. My Code:

```html
<div>
<p> Displaying <strong >{{ $count }} </strong> Elements: </p>

    <button wire:click="more">Load More</button>

</div>
<div>
@for ($i = 1; $i <= $count;$i++)
    <div wire:key-"{{$i}}">
    <p wire:key="loop-{{$i}}"> {{ $i }} </p>
    </div>
@endfor
</div>
```

```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class ElementList extends Component
{
    public $count = 2;

    public function more()
    {
        $this->count += 2;
    }

    public function render()
    {
        return view('livewire.element-list', ['count' => $this->count]);
    }
}
```

Now, interestingly, when clicking on "Load More" the counter does get incremented by two every time i click the button just as it should ! But there only stays two elements that are being displayed. Using Firefox Inspector i also do see that the HTML that is being sent back from the server also has the correct number of <li> - but livewire refuses to rerender the <li> elements for some reason. 

When setting the `public $count = 2` line to a bigger number, they _initial_ rendering is as it should be !

I have used several `:key` `wire:key` variants on each of all this elements, none of them worked.

**Comment by joshhanley** [comment]
@thyseus it looks like you have two root elements in the blade code you supplied. This won't work, try wrapping everything in a div.

I think that is unrelated to the issue reported by @krizzdev though as his issue is related to using Livewire and Vue together.

Can you test what I suggested and if you're still having an issue, can you raise it in it's own issue (unless I'm missing something and it is related to the Vue plugin too)? 🙂

Hope this helps!

**Comment by thyseus** [comment]
@joshhanley thanks a lot for your reply. Your suggestion works. Thanks a lot :). Unfortunately i can not help with this issue furthermore. How about highlighting this "only one <div> as root element in the blade file" more strongly in the documentation? vue.js veterans knows this, beginners like me might stumble into problems.

Update: or what is your opinion about this: https://github.com/livewire/livewire/pull/2342 ?

**Comment by jordyvanderhaegen** [comment]
I've made a [PR](https://github.com/livewire/vue/pull/29) regarding this issue

**Comment by muhammad-tanveerullah** [comment]
Hi there, if someone is still facing issues with this, we can just do a simple fix until we have an actual fix in the livewire itself.

You just have to add and ID to the child component root tag. 

Parent Component:
```
<table>
   <thead></thead>
   <tbody>
        @foreach($users as $key=>$user)
            <livewire:user-info :user="$user :key="$user->id" />
        @endforeach
  </tbody>
<table>

```
Child Component:
```
<tr id="{{$user->id}}">  //Add this and you should see it starts working and you don't have to have div tag as root tag
   <td></td>
<tr>

```

Thanks!!
